File: QTD.SPA/src/app/_DtoModels/Version_Task_Step/Version_Task_Step.ts
import { Entity } from '../Entity';
import { Task_Step } from '../Task_Step/Task_Step';
import { Version_Task } from '../Version_Task/Version_Task';

export class Version_Task_Step extends Entity{
  taskStepId!: any;
  versionTaskId!: any;
  description!: string;
  number!: number;
  version_Task!: Version_Task;
  task_Step!: Task_Step;
}

File: QTD2.Domain/Entities/Core/Version_Task_Step.cs
namespace QTD2.Domain.Entities.Core
{
    public class Version_Task_Step : Common.Entity
    {
        public int TaskStepId { get; set; }

        public int VersionTaskId { get; set; }

        public string Description { get; set; }

        public int? Number { get; set; }

        public virtual Version_Task Version_Task { get; set; }

        public virtual Task_Step Task_Step { get; set; }

        public Version_Task_Step()
        {
        }

        public Version_Task_Step(Task_Step step)
        {
            TaskStepId = step.Id;
            VersionTaskId = step.TaskId;
            Description = step.Description;
            Number = step.Number;
        }

        public Version_Task_Step(Version_Task task, Task_Step step)
        {
            TaskStepId = step.Id;
            VersionTaskId = task.Id;
            Description = step.Description;
            Number = step.Number;
        }
    }
}
